city_scrapers/mixins/mc.py

```diff
@@ -0,0 +1,144 @@
+
+        yield Request(url, callback=self.parse)
+
+    def parse(self, response):
+        """
+        Parse a list of meetings from the response.
+        """
+        print("donkey", response)
+        items = response.json()
+        if not items or len(items) == 0 or "value" not in items:
+            self.logger.warning("No meetings found")
+            return
+        for item in items["value"]:
+            meeting = Meeting(
+                title=item["eventName"],
+                description=item["eventDescription"],
+                classification=self._parse_classification(item["categoryName"]),
+                start=self._parse_start(item["startDateTime"]),
+                end=None,
+                all_day=False,
+                time_notes="",
+                location=self._parse_location(item["eventLocation"]),
+                links=self._parse_links(item),
+                source=response.url,
+            )
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+            yield meeting
+
+    def _parse_classification(self, name):
+        """
+        Parse or generate classification from allowed options.
+        """
+        if "city council" in name.lower():
+            return CITY_COUNCIL
+        if "board" in name.lower():
+            return BOARD
+        if "commission" in name.lower():
+            return COMMISSION
+        if "committee" in name.lower():
+            return COMMITTEE
+        return NOT_CLASSIFIED
+
+    def _parse_start(self, start):
+        """
+        Parse the start date and time.
+        """
+        return datetime.strptime(start, "%Y-%m-%dT%H:%M:%SZ")
+
+    def _parse_location(self, location):
+        """
+        Parse or generate location.
+        """
+        if not location:
+            return {
+                "name": "",
+                "address": "",
+            }
+        # build location from address1, city, state, zipCode
+        address = location["address1"]
+        address_fields = ["address2", "city", "state", "zipCode"]
+        for field in address_fields:
+            if location[field]:
+                address += f", {location[field].strip()}"
+        return {
+            "name": "",
+            "address": address,
+        }
+
+    def _parse_links(self, item):
+        """
+        Parse published files into links.
+        """
+        links = []
+        if "publishedFiles" in item and len(item["publishedFiles"]) > 0:
+            for file in item["publishedFiles"]:
+                links.append(
+                    {
+                        "title": file["name"],
+                        "href": f"https://mandannd.api.civicclerk.com/v1/Meetings/GetMeetingFileStream(fileId={file['fileId']},plainText=false)",  # noqa
```

Comment:
You can replace `https://mandannd.api.civicclerk.com` with `{self.base_url}`

Reply:
Yep. That's def tidier. Thanks, @LienDang.